### ISSUE TITLE
Voice from Server to repeater its broken

### ISSUE THREAD

**Issue by do1kbl** [OK-DMR/Hytera_Homebrew_Bridge#19]
Hello I used the software on the hblink3 server.
Its working finde but the voice its broken.
I test it with hytera RD625 V8.5 and an mmdvm hotspot.
RD625 ---> MMDVM sound very god.
MMDVM---> RD625 its broken.
I test also a Parrot from RD625 
the echo its on mmdvm hotspot fine and on the hytera also broken.
I check yout from here the actual Master

maybe someone knows how I can solve it?

**Comment by kivimartin** [comment]
I have the exact same problem, repeater on v9 code.

**Comment by smarek** [comment]
Yeah, i confirm, my work-life is now quite broken, if you can investigate and fix, i'll be happy to help you with PRs, otherwise, i need some time and i shall get to this in Feb hopefuly

**Comment by kivimartin** [comment]
@smarek Hi! Tnx for quick response. Do you have any indication on where in the code we could start debugging?. Got accesses to repeaters and everything :)

**Comment by smarek** [comment]
I believe order of voice frames is incorrect (or the frames are marked incorrectly in MMDVM/Homebrew protocol, DMRD voice/data and slot bits)
You can compare out and in streams or compare with different hytera/homebrew tools (eg hytera homebrew gateway from oe1kbc or any other)
I dont have more from top of my head, if you find anything, ill appreciate :)

**Comment by smarek** [comment]
test that could solve that

1) take full hytera ipsc -> mmdvm voice transmission
2) pass it through translator to mmdvm
3) pass it through translator to hytera
4) check if the order of voice frames and all necessary protocol markings match the original transmission

i'm currently sick and i lack access to hytera hardware at this location, if anybody wants to create this test-case, we can identify what issues the translator has and fix that

cheers

**Comment by kivimartin** [comment]
I have servers, hytera repeaters and know how to wireshark :) i can get the caputure

**Comment by smarek** [comment]
would be appreciated, thank you @kivimartin

**Comment by ea7ken** [comment]
> I have the exact same problem, repeater on v9 code.

Same Problem With Hytera HR1065 repeater

**Comment by kivimartin** [comment]
Capture 1: From RPT to GW
https://www.cloudshark.org/captures/d84ab4b5dd0a

Capture 2: From GW to RPT (Traffic org from HBLINK)
https://www.cloudshark.org/captures/25105b273f0c

100.101.5.5 = RPT , 172.16.238.10 = GW

If you are availble on discord or any other chat tool, please reach out :)

**Comment by do1kbl** [comment]
Hello How is the status about this ? The files help you or you see something? You need other help?Am 01.01.2025 um 18:57 schrieb Martin Kivi ***@***.***>:﻿
Capture 1: From RPT to GW
https://www.cloudshark.org/captures/d84ab4b5dd0a
Capture 2: From GW to RPT (Traffic org from HBLINK)
https://www.cloudshark.org/captures/25105b273f0c
100.101.5.5 = RPT , 172.16.238.10 = GW
If you are availble on discord or any other chat tool, please reach out :)

**Comment by smarek** [comment]
@kivimartin much appreciated Martin, thanks a lot. I haven't been able to get to the analysis, hopefully this weekend i'll gather enough time and will power.

**Comment by mikitrin** [comment]
Hello. How is the status? I have the same problem (Hytera->HBlink3->MMDVM perfect, MMDVM->HBLink3->Hytera very low quality).

**Comment by ea4hao** [comment]
Hola buenas tardes yo tengo también el problema que de HBLiNK a HHB y a repetidor sale roto .
Se escucha un segundo la voz y ya solo se queda en portadora .
Agradecería ayuda llevo unes para aprender a montar un HHB y solo me falta el paso de la voz rota

**Comment by ea4hao** [comment]
alguien tiene noticias del proyecto

**Comment by dalzazy1337** [comment]
The same thing bro, was it solved?
